pkg/maputil/maputil.go

```diff
@@ -187,5 +188,55 @@ func Set(m map[string]any, key []string, value string) {
 		m, key = getCursor(key[0]).getMap(m), key[1:]
 	}
 
-	getCursor(key[0]).set(m, value)
+	getCursor(key[0]).set(m, typedVal(value, stringBool))
+}
+
+func typedVal(val string, st bool) any {
```

Comment:
please add some comments and unit tests for this function.

Reply:
I've added some tests for all types and added some comments for it.